compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -1312,14 +1312,24 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                 "additional use",
             );
             first_trait.label_with_exp_info(&mut err, "first non-auto trait", "first use");
+            let regular_traits_vec = regular_traits
+                .iter()
+                .map(|t| t.trait_ref().print_only_trait_path().to_string())
+                .collect::<Vec<_>>();
+            let sug_trait_name = if regular_traits_vec.contains(&"NewTrait".to_string()) {
+                let mut append_no = 2;
+                while regular_traits_vec.contains(&format!("NewTrait{append_no}")) {
+                    append_no += 1;
+                }
+                format!("NewTrait{append_no}")
+            } else {
+                "NewTrait".to_string()
+            };
```

Comment:
```suggestion
            let mut regular_traits = regular_traits
                .iter()
                .map(|t| t.trait_ref().print_only_trait_path().to_string());
            let sug_trait_name = regular_traits
                .filter_map(|name| name.strip_prefix("NewTrait")?.parse::<u32>::().ok())
                .max()
                .map_or(|| std::cow::Cow::from("NewTrait", |max| std::cow::Cow::from(format!("NewTrait{}", max + 1)));
```

Several allocations removed :)